docs/excel/custom-functions-parameter-options.md

```diff
@@ -0,0 +1,141 @@
+ * @customfunction 
+ * @param {number} first First number
+ * @param {number} second Second number
+ * @param {number} [third] Third number to add. If omitted, third = 0.
+ * @returns {number} The sum of the two numbers.
+ */
+function add(first, second, third) {
+  if (third) {
+    return first + second + third;
+  }
+  return first + second;
+}
+CustomFunctions.associate("ADD", add);
+```
+
+When you define a function that contains one or more optional parameters, you should specify what happens when the optional parameters are undefined. In the following example, `zipCode` and `dayOfWeek` are both optional parameters for the `getWeatherReport` function. If the `zipCode` parameter is undefined, the default value is set to `98052`. If the `dayOfWeek` parameter is undefined, it is set to Wednesday.
+
+```js
+/**
+ * Gets a weather report for a specified zipCode and dayOfWeek
+ * @customfunction
+ * @param {number} zipCode Zip code
+ * @param {string} dayOfWeek Day of the week
+ * @returns {string} Weather report for the day of the week in that zip code.
+ */
+function getWeatherReport(zipCode, dayOfWeek)
+{
+  if (zipCode === undefined) {
+      zipCode = "98052";
+  }
+
+  if (dayOfWeek === undefined) {
+    dayOfWeek = "Wednesday";
+  }
+
+  // Get weather report for specified zipCode and dayOfWeek.
+  // ...
+}
+```
+
+## Range parameters
+
+Your custom function may accept a range of data as an input parameter. A function can also return a range of data. In JavaScript, a range of data is represented as a two-dimensional array.
+
+For example, suppose that your function returns the second highest value from a range of numbers stored in Excel. The following function accepts the parameter `values`, which is of type `Excel.CustomFunctionDimensionality.matrix`. Note that in the JSON metadata for this function, the parameter's `type` property is set to `matrix`.
+
+```js
+function secondHighest(values){
+  let highest = values[0][0], secondHighest = values[0][0];
+  for(var i = 0; i < values.length; i++){
+    for(var j = 1; j < values[i].length; j++){
+      if(values[i][j] >= highest){
+        secondHighest = highest;
+        highest = values[i][j];
+      }
+      else if(values[i][j] >= secondHighest){
+        secondHighest = values[i][j];
+      }
+    }
+  }
+  return secondHighest;
+}
+```
+
+## Invocation context parameter
+
+Every custom function is automatically passed an `invocation` argument as the last argument, which can be used to find the address of a cell or handle what happens when [canceling a function](custom-functions-web-reqs.md#stream-and-cancel-functions). Even if you declare no parameters, your custom function has this parameter. This argument doesn't appear for a user in Excel. If you want to use `invocation` in your custom function, declare it as the last parameter.
+
+In the following code sample, the `invocation` context is explicitly stated for your reference.
+
+```js
+/**
+ * Add two numbers
+ * @customfunction 
```

Comment:
We should also show the TypeScript version because it is not easy to lookup which type to use for the invocation parameter. Similar to the following...
```typescript
function cellAddress(context: CustomFunctions.Invocation) {
  return context.address;
}
```

Reply:
Is this something we want to do with a toggle between the two or two samples? I thought we were trying to avoid putting in TS in docs in general but I agree it's a lot easier to look at in TS.